gprofiler/profilers/java.py

```diff
@@ -259,11 +288,71 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid}
+        metadata = {
+            "java_version": version,
+            "libjvm_elfid": libjvm_elfid,
+            "jvm_flags": self.get_jvm_flags_serialized(process),
+        }
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    def get_jvm_flags_serialized(self, process: Process) -> Optional[List[Dict]]:
+        jvm_flags = self.get_jvm_flags(process)
+        if jvm_flags is None:
+            return None
+
+        return [flag.to_dict() for flag in sorted(jvm_flags, key=lambda flag: flag.name)]
+
+    def get_jvm_flags(self, process: Process) -> Optional[List[JvmFlag]]:
+        jvm_raw_flags: Optional[List[JvmFlag]] = self.get_jvm_flags_raw(process)
+
+        if jvm_raw_flags is None:
+            return None
+
+        elif self.jvm_flags_to_retrieve == JavaFlagRetrievalOptions.ALL:
```

Comment:
General - `collect` is a more appropriate term here and used in other parts of the codebase. Let's rename this field to `collected_jvm_flags` and `JavaFlagRetrievalOptions` to `JavaFlagsCollectionOptions`